packages/program/src/contract-call-script.ts

```diff
@@ -283,7 +283,7 @@ export const getScriptDataV1: ContractCallScriptFn = (
   // 1. Amount
   scriptData.push(new BigNumberCoder('u64').encode(call.amount || 0));
   // 2. Asset ID
-  scriptData.push(new B256Coder().encode(call.assetId?.toString() || BaseAssetId));
+  scriptData.push(new B256Coder().encode(call.assetId?.toString() || ZeroBytes32));
```

Comment:
We should not use `ZeroBytes32` here as a default value. 

If the used chain config sets the BaseAsset ID to a different value, this will result in an error. 

We need to make the related functions have the base asset ID as a parameter.

Reply:
Could we assume `BaseAssetId` will be the most used? If yes, are there any downsides to defaulting to it contextually, and could this reduce necessary breaking change adjustments?

```ts
call.assetId ?? (call.amount ? BaseAssetId : ZeroBytes32)
```

Reply:
@Torres-ssf let me know what you think of the above or resolve accordingly, as I'll merge after this one is closed 👍🏻 

But as discussed, `ZeroBytes32` is a place holder here for the required property space rather than the value of the asset to forward.

The base asset ID is set on the request level. For an invocation scope, this is actually pretty straightforward and negates breaking changes as we have access to the provider from the program.

Reply:
Base asset ID has been removed from tx request. So if a call is forwarding assets, the asset ID will be passed through to the contract call script. If no assets are being forwards, `ZeroBytes32` is used as the property space placeholder. https://github.com/FuelLabs/fuels-ts/pull/1976/commits/1239a4bbaac8a320ced1c66eeab51224269e0d8d